dotnet/src/webdriver/Firefox/FirefoxProfile.cs

```diff
@@ -296,22 +296,16 @@ private void UpdateUserPreferences()
 
         private void ReadDefaultPreferences()
         {
-            var jsonSerializerOptions = new JsonSerializerOptions
-            {
-                Converters =
-                {
-                    new ResponseValueJsonConverter()
-                }
-            };
-
             using (Stream defaultPrefsStream = ResourceUtilities.GetResourceStream("webdriver_prefs.json", "webdriver_prefs.json"))
             {
                 using (StreamReader reader = new StreamReader(defaultPrefsStream))
                 {
                     string defaultPreferences = reader.ReadToEnd();
-                    Dictionary<string, object> deserializedPreferences = JsonSerializer.Deserialize<Dictionary<string, object>>(defaultPreferences, jsonSerializerOptions);
-                    Dictionary<string, object> immutableDefaultPreferences = deserializedPreferences["frozen"] as Dictionary<string, object>;
-                    Dictionary<string, object> editableDefaultPreferences = deserializedPreferences["mutable"] as Dictionary<string, object>;
+                    JsonObject deserializedPreferences = JsonNode.Parse(defaultPreferences) as JsonObject
```

Comment:
Probably `JsonDocument` would be better.

Reply:
Fixed. It came out much simpler with the `JsonDocument`.